assets/js/modules/pagespeed-insights/components/dashboard/DashboardPageSpeedCTA.js

```diff
@@ -6,23 +24,28 @@ import { __ } from '@wordpress/i18n';
 /**
  * Internal dependencies
  */
+import Data from 'googlesitekit-data';
 import {
 	activateOrDeactivateModule,
 	getReAuthURL,
 	showErrorNotification,
-	getModulesData,
 } from '../../../../util';
 import CTA from '../../../../components/notifications/cta';
 import data from '../../../../components/data';
 import GenericError from '../../../../components/notifications/generic-error';
+import { STORE_NAME as MODULES_STORE } from '../../../../googlesitekit/modules/datastore/constants';
+import { STORE_NAME as USER_STORE, PERMISSION_MANAGE_OPTIONS } from '../../../../googlesitekit/datastore/user/constants';
+const { useSelect } = Data;
 
-const PageSpeedInsightsCTA = () => {
-	const {
-		active,
-		setupComplete,
-	} = getModulesData()[ 'pagespeed-insights' ];
+function DashboardPageSpeedCTA() {
+	const pagespeedInsightsModule = useSelect( ( select ) => select( MODULES_STORE ).getModule( 'pagespeed-insights' ) );
+	const canManageOptions = useSelect( ( select ) => select( USER_STORE ).hasCapability( PERMISSION_MANAGE_OPTIONS ) );
+
+	if ( ! pagespeedInsightsModule ) {
+		return null;
+	}
 
-	const { canManageOptions } = global._googlesitekitLegacyData.permissions;
+	const { active, setupComplete } = pagespeedInsightsModule;
 
 	if ( ! canManageOptions && ! setupComplete ) {
```

Comment:
See my previous comment, let's change this in one of the two ways. Either the component should fully take care of itself, or we should be able to assume it's only used if `! active || ! setupComplete` anyway.

Reply:
Ok, updated it to be `! canManageOptions || ( active && setupComplete )`.